metadata_backend/api/handlers/object.py

```diff
@@ -153,6 +163,13 @@ async def post_object(self, req: Request) -> Response:
                 JSONValidator(content, schema_type).validate
             operator = Operator(db_client)
 
+        schemas_in_submission = set()
+        async for _, schema, _ in self.iter_submission_objects(submission):
```

Comment:
i am not 100% sure but we might be iterating twice over the objects in a submission `get_collection_objects` on line 140 in this file

Reply:
:thinking: it does happen only for `study`, but for me it seems those functions do the same thing: check how many objects of a certain schema are in a submission

Reply:
I changed, so that the iteration has explicitly no calls to DB, which should be cheaper than querying mongo.